src/wesnoth_lua_config.h

```diff
@@ -48,9 +48,6 @@
 /*  Wesnoth definitions for compatibility
  *  No longer needed in C++, included only for Lua runtime.
  */
-#define LUA_COMPAT_5_2
-#define LUA_COMPAT_5_1
-#define LUA_COMPAT_FLOATSTRING
```

Comment:
What were these doing before?

Reply:
See [luaconf.h](https://github.com/lua/lua/blob/12110de/luaconf.h#L291) from before the [code removal](https://github.com/lua/lua/commit/34b00c1).  They control:

* The presence of several deprecated functions in the mathematical library
* The presence of library 'bit32'
* The effectiveness of the __ipairs metamethod
* The presence of macros for manipulating other integer types (lua_pushunsigned, lua_tounsigned, luaL_checkint, luaL_checklong, etc.)
* The presence of global 'unpack' (replaced with 'table.unpack')
* The presence of table 'package.loaders' (replaced with 'package.searchers')
* The macro 'lua_cpcall' to emulate the deprecated function lua_cpcall
* The function 'log10' in the math library
* The function 'loadstring' in the base library
* The function 'maxn' in the table library
* Trivial compatibility for some changes in the API
* Compatibility with previous module functions 'module' (Lua) and 'luaL_register' (C)
* Making Lua format integral floats without a float mark ('.0')

Lua 5.4 no longer has any of this compatibility code, and Wesnoth is apparently working without it.  I just removed the macro definitions to clean up the code and emphasize that they don't preclude using a stock system copy of Lua C++.

Reply:
In other words, you're saying that those macros do nothing in Lua 5.4? They only do something in Lua 5.3?

I think most of the prominent deprecated functions had stubs added by us (to be removed probably in 1.19); the only thing that slightly worries me is `__ipairs`, but if it was already gone, then it should be fine.

Reply:
Right, `LUA_COMPAT_5_2`, `LUA_COMPAT_5_1`, and `LUA_COMPAT_FLOATSTRING` are gone (if you grep Lua 5.4 source code for them, you get nothing), so Wesnoth has been working since upgrading to Lua 5.4 almost two years ago in commit 561cc33 (pull #5535) without the compatibility code those macros enabled.

Although my earlier reply was somewhat incomplete.  Not all of the above listed compatibility code controlled by those old macros is gone.  *Some* of it remains:

* The presence of several deprecated functions in the mathematical library.
* The presence of macros for manipulating other integer types (lua_pushunsigned, lua_tounsigned, luaL_checkint, luaL_checklong, etc.)
* Trivial compatibility for some changes in the API

But again, Wesnoth hasn't been using these features for two years.  And everything else I listed is gone in Lua 5.4.